src/context/hooks/useCartItems.tsx

```diff
@@ -0,0 +1,25 @@
+import { useState } from 'react';
+import { CartItem } from '../../types/CartItem';
+
+type FunctionReturn = [CartItem[], (phonesToSet: CartItem[]) => void];
+
+const KEY = 'cart';
+
+export function useCartItems(initialItems: CartItem[]): FunctionReturn {
+  const [phones, setPhones] = useState<CartItem[]>(() => {
+    try {
+      const storedData = localStorage.getItem(KEY);
+
+      return storedData ? JSON.parse(storedData) : initialItems;
+    } catch (e) {
```

Comment:
You can throw an error in the console just in case for the future

Reply:
I'll do it in next PR, need these changes for now)